auto-configurations/spring-ai-mcp-server/src/main/java/org/springframework/ai/autoconfigure/mcp/server/McpBackwardCompatibility.java

```diff
@@ -44,12 +44,9 @@ public List<BiConsumer<McpSyncServerExchange, List<McpSchema.Root>>> syncRootsCh
 			return List.of();
 		}
 
-		return rootsChangeConsumers.stream().map(c -> new BiConsumer<McpSyncServerExchange, List<McpSchema.Root>>() {
-			@Override
-			public void accept(McpSyncServerExchange exchange, List<McpSchema.Root> roots) {
-				c.accept(roots);
-			}
-		}).collect(Collectors.toList());
+		return rootsChangeConsumers.stream()
+			.map(c -> (BiConsumer<McpSyncServerExchange, List<McpSchema.Root>>) ((exchange, roots) -> c.accept(roots)))
```

Comment:
Would this work?

`.map(c -> (McpSyncServerExchange exchange, List<McpSchema.Root> roots) -> c.accept(roots))`

Reply:
I'm afraid not. Tried it before but it couldn't infer the types (or something like this). 
This code will be removed after spring-ai  M7 milestone release, so it is not that important.